elastic/kibana
# [Fix] Render complete reporting

This PR leaves it up to the individual TSVB / agg based visualization types to call "render complete" in the right place. Some visualizations have a multi-pass rendering which wasn't captured correctly by the `ReactDOM.render()` callback.

Fixed visualizations:
* Lens gauge
* Lens heatmap / new heatmap renderer
* Lens xy
* Lens/Agg based pie
* TSVB timeseries/gauge/metric/topn/markdown/table

To test:
For agg based / TSVB:
* Create a dashboard with a single vis to test
* Add a breakpoint here: https://github.com/elastic/kibana/blob/46edbe1c57c6f35b2295f04ac2f7da471538ed80/src/plugins/visualizations/public/embeddable/visualize_embeddable.tsx#L289
* Reload the page
* Once the breakpoint is hit, the visualization should be fully rendered

For Lens:
* Create a dashboard with a single vis to test
* Add a breakpoint here: https://github.com/elastic/kibana/blob/b29b46896128fabe9b8bc4d727a57192fe76ad8b/x-pack/plugins/lens/public/embeddable/embeddable.tsx#L475
* Reload the page
* Once the breakpoint is hit, the visualization should be fully rendered
